extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultManager.java

```diff
@@ -0,0 +1,60 @@
+package io.quarkus.vault.runtime;
+
+import io.quarkus.vault.runtime.client.OkHttpVaultClient;
+import io.quarkus.vault.runtime.client.VaultClient;
+import io.quarkus.vault.runtime.config.VaultServerRuntimeConfig;
+
+public class VaultManager {
+
+    private static volatile VaultManager instance;
+
+    private VaultServerRuntimeConfig serverConfig;
+    private VaultClient vaultClient;
+    private VaultAuthManager vaultAuthManager;
+    private VaultKvManager vaultKvManager;
+    private VaultDbManager vaultDbManager;
+    private VaultCredentialProvider vaultCredentialProvider;
+
+    public static VaultManager getInstance() {
+        return instance;
+    }
+
+    public static void init(VaultServerRuntimeConfig serverConfig) {
+        if (instance == null) {
+            instance = new VaultManager(serverConfig);
+        }
+    }
+
+    public static void reset() {
+        instance = null;
+    }
+
+    public VaultManager(VaultServerRuntimeConfig serverConfig) {
+        this(serverConfig, new OkHttpVaultClient(serverConfig));
```

Comment:
You don't need to close the client somehow?

Reply:
I just null out the vault manager singleton in `VaultManager.reset()`
I searched how to do a proper close on okhttpclient, and found this: [shutdown is not necessary](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/#shutdown-isnt-necessary)